tests/test_training/test_training_loop.py

```diff
@@ -2,33 +2,86 @@
 
 """Test for sLCWA and LCWA."""
 
+from pykeen.losses import MarginRankingLoss, NSSALoss, SoftplusLoss
 from pykeen.sampling.filtering import BloomFilterer, PythonSetFilterer
 from pykeen.training import LCWATrainingLoop, SLCWATrainingLoop
 from tests.test_training import cases
 
 
-class UnfilteredSLCWATrainingLoopTestCase(cases.SLCWATrainingLoopTestCase):
-    """Test sLCWA with unfiltered negative sampling."""
+class MRUnfilteredSLCWATrainingLoopTestCase(cases.SLCWATrainingLoopTestCase):
+    """Test sLCWA with unfiltered negative sampling with margin ranking loss."""
 
     cls = SLCWATrainingLoop
     filterer = None
+    loss_cls = MarginRankingLoss
 
 
-class SetFilteredSLCWATrainingLoopTestCase(cases.SLCWATrainingLoopTestCase):
-    """Test sLCWA with set filtered negative sampling."""
+class NSSAUnfilteredSLCWATrainingLoopTestCase(cases.SLCWATrainingLoopTestCase):
+    """Test sLCWA with unfiltered negative sampling with NSSA loss."""
+
+    cls = SLCWATrainingLoop
+    filterer = None
+    loss_cls = NSSALoss
+
+
+class SoftplusUnfilteredSLCWATrainingLoopTestCase(cases.SLCWATrainingLoopTestCase):
+    """Test sLCWA with unfiltered negative sampling with softplus loss."""
+
+    cls = SLCWATrainingLoop
+    filterer = None
+    loss_cls = SoftplusLoss
+
+
+class MRSetFilteredSLCWATrainingLoopTestCase(cases.SLCWATrainingLoopTestCase):
+    """Test sLCWA with set filtered negative sampling with margin ranking loss."""
+
+    cls = SLCWATrainingLoop
+    filterer = PythonSetFilterer
```

Comment:
consistency: this should be `filterer_cls`?

Reply:
Yes you’re right

Reply:
But also we should change the interface for the model to take a class instead of an instance

Reply:
At least on the level of the negative sampler we already support giving `Hint`s for the filterer.

Reply:
Done in https://github.com/pykeen/pykeen/pull/462/commits/447a4513c02a21a349f2428842db069e5e7bf7a4